pkg/endpoint/endpoint_status.go

```diff
@@ -172,88 +172,123 @@ func updateLabels(allocator cache.IdentityAllocator, allowedIdentityTuple *ciliu
 	}
 }
 
-// populateResponseWithPolicyKey inserts an AllowedIdentityTuple element into 'policy'
-// which corresponds to the specified 'desiredPolicy'.
-func populateResponseWithPolicyKey(allocator cache.IdentityAllocator, policy *cilium_v2.EndpointPolicy, policyKey *policy.Key) {
-	allowedIdentityTuple := cilium_v2.AllowedIdentityTuple{
+// populateResponseWithPolicyKey inserts an AllowedIdentityTuple element into
+// 'policy' which corresponds to the specified 'desiredPolicy'. If 'isDeny' is
+// true, it will insert the policyKey into the 'Denied'.
+func populateResponseWithPolicyKey(
+	allocator cache.IdentityAllocator,
+	policy *cilium_v2.EndpointPolicy,
+	policyKey *policy.Key,
+	isDeny bool,
+) {
+	identityTuple := cilium_v2.IdentityTuple{
 		DestPort: policyKey.DestPort,
 		Protocol: policyKey.Nexthdr,
 		Identity: uint64(policyKey.Identity),
 	}
 
 	secID := identity.NumericIdentity(policyKey.Identity)
-	updateLabels(allocator, &allowedIdentityTuple, secID)
+	updateLabels(allocator, &identityTuple, secID)
 
 	switch {
 	case policyKey.IsIngress():
-		if policy.Ingress.Allowed == nil {
-			policy.Ingress.Allowed = cilium_v2.AllowedIdentityList{allowedIdentityTuple}
+		if !isDeny {
+			if policy.Ingress.Allowed == nil {
+				policy.Ingress.Allowed = cilium_v2.AllowedIdentityList{identityTuple}
+			} else {
+				policy.Ingress.Allowed = append(policy.Ingress.Allowed, identityTuple)
+			}
```

Comment:
Nit: we don't need to negate `isDeny`. We can just invert these cases. Makes it easier to read first time without needing to do the mental computation :)